### PR TITLE
Disable SIP/Prefill for non-test accounts

### PR DESCRIPTION
**Pull request by markolson**
Repurposing some work I started last night to gate things to just LOA3 to do that gate, but also add one that limits to obviously fake SSNs that we use for testing.

This is meant to disable this feature for any non-vets.gov test users, as the 796-*-* ssn pattern is not used for actual ssns.

### CODE REVIEW

**Comment by robbiethegeek-usds** [comment]
This is really starting to feel brittle, that is my first thought the code makes sense just worried about adding code that won't be used in prod and it is going to be on prod servers

File: app/serializers/user_serializer.rb [hunk 1]
```diff
@@ -89,6 +89,9 @@ def services
     service_list << BackendServices::USER_PROFILE if object.can_access_user_profile?
     service_list << BackendServices::APPEALS_STATUS if beta_enabled?(object.uuid, 'appeals_status') &&
                                                        object.can_access_appeals?
+
+    service_list << BackendServices::SAVE_IN_PROGRESS if object.can_save_partial_forms?
```
Comment (omgitsbillryan) [inline app/serializers/user_serializer.rb]: `service_list` represents features within Vets.gov that a user has access to.  `vets-website` consumes this list as a means to either block or allow a given user into a certain area of the Vets.gov site.

`BackendServices::SAVE_IN_PROGRESS` and `BackendServices::FORM_PREFILL`  _mostly_ fits that description but I wanted to point out it's kind of like apples to pears.  That said, I'm OK with it.

File: app/models/user.rb [hunk 2]
```diff
@@ -78,6 +78,14 @@ def can_access_appeals?
     loa3? && ssn.present?
   end
 
+  def can_save_partial_forms?
+    loa3? && ssn.present? && ssn.starts_with?('796')
```
Comment (omgitsbillryan) [inline app/models/user.rb]: '796' is a magic number.  Can we use a named constant instead?

**Comment by markolson** [comment]
Maybe I can add like a 
```features:
  prefill: false
  save_in_progress: false```
to settings.yml and globally disable it for prod, but let dev/staging override? The code needs to be in so that we can develop and test against it locally and in staging though..

File: spec/request/in_progress_forms_request_spec.rb [hunk 3]
```diff
@@ -20,12 +21,32 @@
   end
 
   describe '#index' do
+    let(:user) { loa3_user }
```
Comment (omgitsbillryan) [inline spec/request/in_progress_forms_request_spec.rb]: Is this necessary, can't you just use the previously defined `loa3_user`?

Comment (markolson) [inline spec/request/in_progress_forms_request_spec.rb]: `:user` is set to be `loa1_user` in a other spec, letting `auth_header` switch between the two without any additional code.

**Pull request merged**